feat: modified SWEBench modal harness to use modal experimental snapshots and processing TestOutput

clee-codegen opened the pull request:
# Motivation

<!-- Why is this change necessary? -->

# Content

<!-- Please include a summary of the change -->

# Testing

<!-- How was the change tested? -->

# Please check the following before marking your PR as ready for review

- [ ] I have added tests for my changes
- [ ] I have updated the documentation or added new documentation as needed

clee-codegen committed:
convert to modal.Dict snapshot manager

clee-codegen committed:
fix: implement modified swebench harness evaluation

clee-codegen committed:
Automated pre-commit update

jemeza-codegen commented on codegen-examples/examples/swebench_agent_run/run_swebench_modal_harness.py:
```diff
@@ -0,0 +1,300 @@
+        self._stream_tasks = []
+
+        # Hack for pylint
+        self.write_file("/sys/fs/cgroup/cpu/cpu.shares", "2048")
```
could we call `super().init(...)` here?

clee-codegen replied:
the init builds the image eagerly, which I thought we might not want to do if we have a snapshot available. maybe modal will cache it correctly and it'd be fine? 🤷 But if modal image caching works then, we don't really need an additional snapshotting solution on top of that right? since all the dependencies get installed in this image.. 

```py
class ModalSandboxRuntime:
    """
    Runtime for running instances in a Modal Sandbox.
    """

    def __init__(
        self, test_spec: TestSpec, timeout: int | None = None, verbose: bool = True
    ):
        self.test_spec = test_spec
        self.image = ModalSandboxRuntime.get_instance_image(test_spec)
        self.sandbox = self._get_sandbox(timeout)
        self.verbose = verbose
        self._stream_tasks = []

```

jemeza-codegen replied:
I see, I don't modal is caching it right now, what you have is good! I missed the eager image creation

jemeza-codegen commented on codegen-examples/examples/swebench_agent_run/snapshot_manager.py:
```diff
@@ -0,0 +1,57 @@
+
+    def get_snapshot_uid(self, example: SWEbenchInstance) -> str:
+        snapshot_meta = self.read_snapshot_meta()
+        return snapshot_meta[example.repo][example.base_commit]
```
`example.base_commit` -> `example.environment_setup_commit`

jemeza-codegen commented on codegen-examples/examples/swebench_agent_run/snapshot_manager.py:
```diff
@@ -0,0 +1,57 @@
+        raise NotImplementedError("Not implemented")
+
+
+class VolumeSnapshotManager(SnapshotManager):
```
would we still need this?

clee-codegen replied:
Nah, we'd only use the ModalDict implementation. I left in here if for whatever reason modal Dict didn't work out. I saw in the documentation that its not a very stable dict in that it could get lost.

jemeza-codegen reviewed (commented):
Looks like a good start!

clee-codegen committed:
base_commit -> environment_setup_commit

clee-codegen committed:
feat: codegen parse oss repos via CLI and modal (#545)

# Motivation

The **Codegen on OSS** package provides a pipeline that:

- **Collects repository URLs** from different sources (e.g., CSV files
or GitHub searches).
- **Parses repositories** using the codegen tool.
- **Profiles performance** and logs metrics for each parsing run.
- **Logs errors** to help pinpoint parsing failures or performance
bottlenecks.

<!-- Why is this change necessary? -->

# Content

<!-- Please include a summary of the change -->
see
[codegen-on-oss/README.md](https://github.com/codegen-sh/codegen-sdk/blob/acfe3dc07b65670af33b977fa1e7bc8627fd714e/codegen-on-oss/README.md)

# Testing

<!-- How was the change tested? -->
`uv run modal run modal_run.py`
No unit tests yet 😿 

# Please check the following before marking your PR as ready for review

- [ ] I have added tests for my changes
- [x] I have updated the documentation or added new documentation as
needed

clee-codegen committed:
add: integrate with postgresql output

clee-codegen committed:
Automated pre-commit update

clee-codegen committed:
Merge branch 'develop' into swebench-sandbox-snapshots

clee-codegen committed:
wip: integration

clee-codegen committed:
fix: integration with modal deployments

clee-codegen committed:
wip: initial refactor

clee-codegen committed:
fix: refactor run to complete

clee-codegen committed:
Merge remote-tracking branch 'origin/develop' into swebench-sandbox-snapshots

clee-codegen committed:
wip: merge changes from run_eval develop

clee-codegen committed:
add: coarse retries for agent run

clee-codegen committed:
fix: limit agent modal function concurrency

clee-codegen committed:
fix: post-merge   bugs

clee-codegen committed:
Merge branch 'develop' into swebench-sandbox-snapshots

clee-codegen committed:
Merge branch 'develop' into swebench-sandbox-snapshots

clee-codegen committed:
Merge branch 'develop' into swebench-sandbox-snapshots

clee-codegen committed:
Merge remote-tracking branch 'origin/develop' into swebench-sandbox-snapshots

clee-codegen committed:
fix: end-to-end to metrics

clee-codegen committed:
Merge remote-tracking branch 'origin/develop' into swebench-sandbox-snapshots
